integration-tests/google-pubsub/src/test/java/org/apache/camel/quarkus/component/google/pubsub/it/GooglePubsubTest.java

```diff
@@ -56,6 +74,219 @@ public void jacksonSerializer() {
+                .then()
+                .statusCode(201);
+
+        Set<String> results = new HashSet<>();
+        results.add(RestAssured.given()
+                .get("/google-pubsub/receive/subscription/google-pubsub.grouped-subscription-name")
+                .then()
+                .statusCode(200).extract().asString());
+
+        results.add(RestAssured.given()
+                .get("/google-pubsub/receive/subscription/google-pubsub.grouped-subscription-name")
+                .then()
+                .statusCode(200).extract().asString());
+
+        Assertions.assertTrue(results.contains("body1"));
+        Assertions.assertTrue(results.contains("body2"));
+
+    }
+
+    //Disabled on real account because of https://issues.apache.org/jira/browse/CAMEL-18277
+    @DisabledIfEnvironmentVariable(named = "GOOGLE_APPLICATION_CREDENTIALS", matches = ".+")
+    // https://github.com/apache/camel-quarkus/issues/3944
+    @DisabledIfEnvironmentVariable(named = "CI", matches = "true")
+    @Test
+    @Order(4)
+    public void testOrdering() throws Exception {
+        LOG.info("Start of the ordering test");
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ORDERING_DIRECT_IN)
+                .body("1")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ORDERING_DIRECT_IN)
+                .body("2")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ORDERING_DIRECT_IN)
+                .body("3")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ORDERING_DIRECT_IN)
+                .body("4")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ORDERING_DIRECT_IN)
+                .body("5")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        RestAssured.given()
+                .queryParam("toEndpoint", GooglePubSubRoutes.ORDERING_DIRECT_IN)
+                .body("6")
+                .post("/google-pubsub/sendToEndpoint")
+                .then()
+                .statusCode(201);
+        LOG.info("All messages were sent");
+
+        await().atMost(5, TimeUnit.SECONDS).until(() -> RestAssured.given()
+                .get("/google-pubsub/receive/subscriptionOrdering/google-pubsub.ordering-subscription-name")
+                .then()
+                .statusCode(200)
```

Comment:
Maybe this line is redundant ?

Reply:
Same for other lines below.

Reply:
Yes, duplicated status are there by mistake. Removed.